src/control/cmd/daos/pool.go

```diff
@@ -337,6 +338,55 @@ func queryPool(poolHdl C.daos_handle_t, queryMask daos.PoolQueryMask) (*daos.Poo
 	return poolInfo, nil
 }
 
+func queryPool(poolHdl C.daos_handle_t, queryMask daos.PoolQueryMask) (*daos.PoolInfo, error) {
```

Comment:
do these really need to be done sequentially, can't the mask handle multiple options set in the same query?

Reply:
We could do that in the long-term goal, introduce daos_pool_query_v2() to return three rank lists at the same time.